### PR TITLE
Separate pipeline for dotnet 5 test suite

### PR DESCRIPTION
**Pull request by vtikoo**
Signed-off-by: Vikas Tikoo <vitikoo@microsoft.com>

### CODE REVIEW

File: .azure_pipelines/ci-pipeline-dotnet-test-suite.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - group: myst-pipeline
+  - name: SCRIPTS_ROOT
+    value: $(Build.SourcesDirectory)/.azure_pipelines/scripts
+
+# dotnet tests pipeline will be triggered in pipeline schedule settings.
+trigger: none
+pr: none
+
+jobs:
+  - job: BuildAndTest
```
Comment (salsal97) [inline .azure_pipelines/ci-pipeline-dotnet-test-suite.yml:11]: Theres an extra space here before the '-' that az pipelines might complain about

Comment (salsal97) [reply .azure_pipelines/ci-pipeline-dotnet-test-suite.yml:11]: The entire thing would accordingly go forward by 1 space.. (this is the syntax mistake I always make too)

Comment (vtikoo) [reply .azure_pipelines/ci-pipeline-dotnet-test-suite.yml:11]: The indentation here is similar to ci-pipelines-makefile.yml, which indents the jobs list by 2 spaces.

Comment (salsal97) [reply .azure_pipelines/ci-pipeline-dotnet-test-suite.yml:11]: Gotcha, I was looking at [this](https://github.com/deislabs/mystikos/blob/main/.azure_pipelines/ci-pipeline-makefile-nightly.yml). Should work then.

File: solutions/coreclr/Makefile [hunk 2]
```diff
@@ -33,7 +33,7 @@ ext2fs: appdir
 	$(MYST) fssig --roothash ext2fs > roothash
 
 run:
-	./test-runner.sh $(MYST_EXEC) ext2 $(TEST_FILE)
+	COMPlus_EnableDiagnostics=0 COMPlus_EnableAlternateStackCheck=1 ./test-runner.sh $(MYST_EXEC) ext2 $(TEST_FILE)
```
Comment (jxyang) [inline solutions/coreclr/Makefile:36]: I thought we block the environment variables from host by default, unless we explicitly allow it in config.json. Did we change the behavior recently?

Comment (vtikoo) [reply solutions/coreclr/Makefile:36]: I tested with a simple program. It does look like passing host env variables is the default.

Comment (jxyang) [reply solutions/coreclr/Makefile:36]: I think that's a regression then. In the [document](https://github.com/deislabs/mystikos/blob/main/doc/sign-package.md) it says:

>  HostEnvironmentVariables | A list of environment variables that can be imported from the insecure host

Without an explicit entry in config.json that's measured, we shouldn't rely on any environment variable from the untrusted host, even though that's convenient. 

The remedy here is to create a config.json and put these two variables in there and run myst with `--app-config-path`

Comment (jxyang) [reply solutions/coreclr/Makefile:36]: Please create an issue for the regression.

Comment (vtikoo) [reply solutions/coreclr/Makefile]: > The remedy here is to create a config.json and put these two variables in there and run myst with --app-config-path

Fixed.

**Review by jxyang — APPROVED** [review]

**Pull request merged**